class/UamAccessHandler.php

```diff
@@ -440,8 +436,8 @@ public function checkObjectAccess($sObjectType, $iObjectId)
                     ) {
                         $this->_aObjectAccess[$sObjectType][$iObjectId] = true;
                         break;
-                    } elseif ($this->getUserAccessManager()->atAdminPanel() && $oUserGroup->getWriteAccess() == 'all'
-                        || !$this->getUserAccessManager()->atAdminPanel() && $oUserGroup->getReadAccess() == 'all'
+                    } elseif (($this->getUserAccessManager()->atAdminPanel() && $oUserGroup->getWriteAccess() == 'all')
```

Comment:
Braces aren't necessary for && because && has a higher priority than ||, see [http://php.net/manual/fa/language.operators.precedence.php](http://php.net/manual/fa/language.operators.precedence.php)

So (A && B) || (C && D) is equivalent to A && B || C && D